test/kmedoids.jl

```diff
@@ -30,14 +30,14 @@ R = kmedoids(costs, k)
 # group 3: [6, 8, 9], values: [21, 20, 22]
 #
 
-X = reshape(@compat(map(Float64, [1, 6, 2, 3, 7, 21, 8, 20, 22])), 1, 9)
+X = reshape(map(Float64, [1, 6, 2, 3, 7, 21, 8, 20, 22]), 1, 9)
 costs = pairwise(SqEuclidean(), X)
 
 R = kmedoids!(costs, [1, 2, 6])
 @test isa(R, KmedoidsResult)
 @test R.medoids == [3, 5, 6]
 @test R.assignments == [1, 2, 1, 1, 2, 3, 2, 3, 3]
 @test R.counts == [3, 3, 3]
-@test_approx_eq R.acosts [1, 1, 0, 1, 0, 0, 1, 1, 1]
-@test_approx_eq R.totalcost 6.0
+@test all(isapprox.(R.acosts, [1, 1, 0, 1, 0, 0, 1, 1, 1]))
```

Comment:
All the `@compat(Union{..})` are also not supported on 0.4. At this point, I wouldn't care about 0.4. And it's not like the new Clustering.jl added many new features 0.4 users may want to use. :-)